cmd/demo/demo.go

```diff
@@ -0,0 +1,340 @@
+				prev, _ := l.Client.GetSecretManagerSecretValue(spec.Source.Project, spec.Source.Secret)
+				next, _ := strconv.Atoi(string(prev))
+
+				// secret value swaps between 0 and 1 every SwitchPeriod
+				next = 1 - next
+				l.Client.UpsertSecretManagerSecret(spec.Source.Project, spec.Source.Secret, []byte(strconv.Itoa(next)))
+			}
+		}
+	}
+}
+
+func (d *logData) Append(t time.Duration, gsm []byte, k8s []byte) {
+	d.Time = append(d.Time, t.Seconds())
+	d.GSMSecretLog = append(d.GSMSecretLog, string(gsm))
+	d.K8sSecretLog = append(d.K8sSecretLog, string(k8s))
+}
+
+func (d *logData) TextReport() string {
+	msg := ""
+	for i := range d.Time {
+		if i == 0 {
+			msg += fmt.Sprintf("\t[%v] K8s secret value intial value: '%s'\n", time.Duration(d.Time[i])*time.Second, d.K8sSecretLog[i])
+			msg += fmt.Sprintf("\t[%v] GSM secret value intial value: '%s'\n", time.Duration(d.Time[i])*time.Second, d.GSMSecretLog[i])
+		} else {
+			if d.GSMSecretLog[i] != d.GSMSecretLog[i-1] {
+				msg += fmt.Sprintf("\t[%v] GSM secret value updated from '%s' to '%s'\n", time.Duration(d.Time[i])*time.Second, d.GSMSecretLog[i-1], d.GSMSecretLog[i])
+			}
+			if d.K8sSecretLog[i] != d.K8sSecretLog[i-1] {
+				msg += fmt.Sprintf("\t[%v] K8s secret value updated from '%s' to '%s'\n", time.Duration(d.Time[i])*time.Second, d.K8sSecretLog[i-1], d.K8sSecretLog[i])
+			}
+		}
+
+	}
+	return msg
+}
+
+// Plot plots the timeline of secret values of both source and destination
+func (d *logData) Plot(name string) {
+
+	p, err := plot.New()
+	if err != nil {
+		klog.Error(err)
+	}
+
+	p.Title.Text = "Secret Sync Timelime"
+	p.X.Label.Text = "time (sec.)"
+	p.Y.Label.Text = "secret value"
+
+	n := len(d.Time)
+	ptsGSM := make(plotter.XYs, n)
+	ptsK8s := make(plotter.XYs, n)
+	for i := 0; i < n; i++ {
+		ptsGSM[i].X = d.Time[i]
+		val, _ := strconv.ParseFloat(d.GSMSecretLog[i], 64)
+		ptsGSM[i].Y = val
+
+		ptsK8s[i].X = d.Time[i]
+		val, _ = strconv.ParseFloat(d.K8sSecretLog[i], 64)
+		ptsK8s[i].Y = val
+	}
+
+	gsm, err := plotter.NewLine(ptsGSM)
+	if err != nil {
+		klog.Error(err)
+	}
+
+	gsm.LineStyle.Width = vg.Points(1)
+	gsm.LineStyle.Color = color.RGBA{R: 255, A: 255}
+
+	k8s, err := plotter.NewLine(ptsK8s)
+	if err != nil {
+		klog.Error(err)
+	}
+
+	k8s.LineStyle.Width = vg.Points(1)
+	k8s.LineStyle.Dashes = []vg.Length{vg.Points(5), vg.Points(5)}
+	k8s.LineStyle.Color = color.RGBA{B: 255, A: 255}
+
+	p.Add(gsm, k8s)
+	p.Legend.Add("Secret Manager secret", gsm)
+	p.Legend.Add("Kubernetes secret", k8s)
+
+	// add margins for better visualization
+	p.X.Min = -1
+	p.X.Max = d.Time[n-1] + 1
+	p.Y.Min = -1
+	p.Y.Max = 2
+
+	if err := p.Save(4*vg.Inch, 4*vg.Inch, name); err != nil {
+		klog.Error(err)
+	}
+
+}
+
+// ShowResults shows the timelines in text form for all secret sync pairs.
+// It also outputs a plot "timeline_i" for the ith pair.
+func (l *SecretSyncLogger) ShowResults() {
+	for i, spec := range l.Agent.Config().Specs {
+		name := fmt.Sprintf("timeline_%d.png", i)
```

Comment:
For followup: can you use a flag to specify a path where these files should land?

Reply:
added a flag `output-path`